server/src/main/java/org/elasticsearch/index/query/CoordinatorRewriteContext.java

```diff
@@ -99,4 +108,8 @@ public IndexLongFieldRange getFieldRange(String fieldName) {
     public CoordinatorRewriteContext convertToCoordinatorRewriteContext() {
         return this;
     }
+
+    public String tier() {
+        return tier;
+    }
```

Comment:
I believe we need the `tier` as the `CoordinatorRewriteContext` doesn't know of `indexSettings` otherwise https://github.com/elastic/elasticsearch/pull/114990/files/67f82e9aa742e223de4cd6a76d496ac66987130e#diff-45c9ed5af87237215a5a11f3d2549aa678272c2cbc898b866b43516ec992fab5R60
The index settings would be needed for us to parse the `_tier_preference` setting.

I'm not sure if it should ? Or if we should keep just passing the `tier` ?

Reply:
I was under the impression that that is the same value that the constant field type that the context exposes returns, or is that something else?

Reply:
The field type does not store any state but every time it evaluates it reaches out to the index settings for the `_tier_preference` value https://github.com/elastic/elasticsearch/pull/114990/files/67f82e9aa742e223de4cd6a76d496ac66987130e#diff-e776fbfb85ffa7c76ad9fbbd1263953047589070bf15add46298deaf929778dbR95

The `QueryWriteContext` stores the index settings (but the `CoordinatorRewriteContext` doesn't)

Reply:
sounds good, I had misunderstood.